q2_metadata/normalization/_norm_rules.py

```diff
@@ -0,0 +1,332 @@
+                'ontology': None,
+                'expected': []
+            },
+            'allowed': {
+                'blank': None,
+                'missing': None
+            },
+            'format': None
+        }
+        self.parsed_rules = self.parse_rule(variable_rules_fp)
+        self.warnings = WarningsCollection()
+        self.errors = ErrorsCollection()
+
+    @staticmethod
+    def parse_rule(variable_rules_fp: str):
+        """Read the yaml file of the current variable's rules.
+
+        Parameters
+        ----------
+        variable_rules_fp : str
+            Path to one variable's rules yaml file.
+
+        Returns
+        -------
+        parsed_rules : dict
+            Parsed variable's rules.
+
+        """
+        with open(variable_rules_fp) as handle:
+            try:
+                parsed_rules = yaml.load(handle, Loader=yaml.FullLoader)
+                return parsed_rules
+            except yaml.YAMLError:
+                raise IOError("Something is wrong with the .yml rule file.")
+            except yaml.scanner.ScannerError:
+                raise IOError("Something is wrong with the .yml rule file.")
+            except UnicodeDecodeError:
+                raise IOError("Something is wrong with the .yml rule file.")
+
+    def normalize(self, variable: str, input_column: pd.Series) -> pd.Series:
+        """
+        Dispatch the metadata column to be curated to
+        the functions that will be applied to properly process
+        the current variable according to the current set of
+        rules for it.
+
+        Parameters
+        ----------
+        variable : str
+            Name of the variable which rules are parsed and
+            that is the yaml file name without the .yml extension.
+        input_column : pd.Series
+            Metadata column to be curated (that may already got
+            through this function).
+
+        Returns
+        -------
+        output_column : pd.Series
+            Curated (or being curated) metadata column.
+
+        """
+        pass
+
+
+class RulesCollection(object):
+    """Collect all the rules for all the variables.
+
+    Parameters
+    ----------
+    variables_rules_dir : str
+        Path to the folder containing the variables rules in yaml.
+
+    Attributes
+    ----------
+    variables_rules_files : list
+        Paths to the yaml rules files.
+    warnings : WarningsCollection instance
+        Collection of warnings encountered during either the parsing
+        of the variables rules, or during the application the rules.
+    errors : ErrorsCollection instance
+        Collection of errors encountered during either the parsing
+        of the variables rules, or during the application the rules.
+
+    Methods
+    -------
+    check_variables_rules
+        Fills `rules` attribute with correctly formatted rules
+        for the current variable, or raise user-defined exception.
+    _check_variables_rules_dir
```

Comment:
Remember to do a global find/replace of _check_variables_rules_dir to check_variables_rules_dir

Reply:
accounted for - will be part of upcoming commit